processor/src/main/java/com/blarley/rxretrobusseed/annotationprocessor/processor/RxRetroBusAnnotationProcessor.java

```diff
@@ -150,22 +159,31 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
         StringBuilder constructorDefinition = new StringBuilder();
 
         // Append the instance fields
-        for (String generatedClass: generatedClasses) {
-            String[] str = generatedClass.split("_");
+        for (GeneratedClass generatedClass : generatedClasses) {
+            String[] str = generatedClass.getName().split("_");
             String baseClassName = str[1];
             clientsFile.append(
                     "\tpublic "
-                            + generatedClass
+                            + generatedClass.getName()
                             + " "
                             + baseClassName
                             + ";\n");
 
-            constructorDefinition.append(
-                    "\t\tthis."
-                            + baseClassName
-                            + " = new "
-                            + generatedClass
-                            + "(retrofitBuilder, bus);\n");
+            if (generatedClass.isRetrofitEnabled()) {
+                constructorDefinition.append(
+                        "\t\tthis."
+                                + baseClassName
+                                + " = new "
+                                + generatedClass.getName()
+                                + "(retrofitBuilder, bus);\n");
```

Comment:
same as the else block save the last line
only thing that needs to be inside of if else is a string that is either (retrofitBuilder, bus);\n or (bus);\n